openwisp_monitoring/device/models.py

```diff
@@ -19,6 +20,21 @@ class Meta:
         proxy = True
         swappable = swappable_setting('device_monitoring', 'DeviceData')
 
+    def handle_unknown_status_change(self, instance, **kwargs):
+        critical_metrics = settings.OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS
+        if instance.metric.name in critical_metrics:
+            if kwargs.get('created', False):
+                self.update_status('unknown')
+            elif kwargs.get('sender', None).__name__ == 'Check':
+                self.update_status('unknown')
+
+    def handle_critical_check_change(cls, check):
+        critical_metrics = settings.OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS
+        device_data_instances = DeviceData.objects.filter(pk=check.object_id)
+        for instance in device_data_instances:
+            if check.metric.name in critical_metrics:
+                instance.update_status('unknown')
```

Comment:
We don't add any code directly to the Model class. We add all the logic to the AbstractModel, so it allows to extend the models. Please keep this in mind.